batch/batch/front_end/front_end.py

```diff
@@ -250,17 +293,16 @@ async def _query_batch_jobs(request, batch_id):
 
 @routes.get('/api/v1alpha/batches/{batch_id}/jobs')
 @prom_async_time(REQUEST_TIME_GET_JOBS)
-@rest_authenticated_users_only
-async def get_jobs(request, userdata):
+@rest_billing_project_users_only
+async def get_jobs(request, userdata):  # pylint: disable=unused-argument
     batch_id = int(request.match_info['batch_id'])
```

Comment:
The fas that `rest_billing_project_users_only` does not pass the batch_id to get_jobs means that a refactoring can easily break our protection mechanisms. Let's make both `_billing_project_users_only` pass the approved batch_id as an argument to the wrapped method.

Reply:
Can you give an example?

Reply:
Like this:
```
@rest_billing_project_users_only
 async def get_jobs(request, userdata, batch_id):  # pylint: disable=unused-argument
```
```
async def user_can_access(db: Database, batch_id: int, username: str):
     return None is not await db.select_and_fetchone(
         '''
 SELECT id
 FROM batches
 LEFT JOIN billing_project_users ON batches.billing_project = billing_project_users.billing_project
 WHERE id = %s AND billing_project_users.`user` = %s;
 ''',
         (batch_id, username))

def rest_billing_project_users_only(fun):
     @rest_authenticated_users_only
     @wraps(fun)
     async def wrapped(request, userdata, *args, **kwargs):
         batch_id = request['batch_id']
         if not await user_can_access(request['db'], batch_id, userdata['username']):
             raise web.HTTPNotFound()
        return await fun(request, userdata, batch_id, *args, **kwargs)
     return wrapped
```